internal/controller/controllers/agentserviceconfig_controller.go

```diff
@@ -592,52 +601,23 @@ func (r *AgentServiceConfigReconciler) newAssistedServiceDeployment(instance *ai
 		// local auth secret
 		newSecretEnvVar("EC_PUBLIC_KEY_PEM", "ec-public-key.pem", agentLocalAuthSecretName),
 		newSecretEnvVar("EC_PRIVATE_KEY_PEM", "ec-private-key.pem", agentLocalAuthSecretName),
+	}
 
-		// image overrides
-		{Name: "AGENT_DOCKER_IMAGE", Value: AgentImage()},
-		{Name: "CONTROLLER_IMAGE", Value: ControllerImage()},
-		{Name: "INSTALLER_IMAGE", Value: InstallerImage()},
-		{Name: "SELF_VERSION", Value: ServiceImage()},
-		{Name: "OPENSHIFT_VERSIONS", Value: OpenshiftVersions()},
-
-		{Name: "ISO_IMAGE_TYPE", Value: "minimal-iso"},
-		{Name: "S3_USE_SSL", Value: "false"},
-		{Name: "LOG_LEVEL", Value: "info"},
-		{Name: "LOG_FORMAT", Value: "text"},
-		{Name: "INSTALL_RH_CA", Value: "false"},
-		{Name: "REGISTRY_CREDS", Value: ""},
-		{Name: "DEPLOY_TARGET", Value: "k8s"},
-		{Name: "STORAGE", Value: "filesystem"},
-		{Name: "ISO_WORKSPACE_BASE_DIR", Value: "/data"},
-		{Name: "ISO_CACHE_DIR", Value: "/data/cache"},
-
-		// from configmap
-		{Name: "AUTH_TYPE", Value: "local"},
-		{Name: "BASE_DNS_DOMAINS", Value: ""},
-		{Name: "CHECK_CLUSTER_VERSION", Value: "False"},
-		{Name: "CREATE_S3_BUCKET", Value: "False"},
-		{Name: "ENABLE_KUBE_API", Value: "True"},
-		{Name: "ENABLE_SINGLE_NODE_DNSMASQ", Value: "True"},
-		{Name: "IPV6_SUPPORT", Value: "True"},
-		{Name: "JWKS_URL", Value: "https://api.openshift.com/.well-known/jwks.json"},
-		{Name: "PUBLIC_CONTAINER_REGISTRIES", Value: "quay.io,registry.svc.ci.openshift.org"},
-		{Name: "WITH_AMS_SUBSCRIPTIONS", Value: "False"},
-
-		{
-			Name: "NAMESPACE",
-			ValueFrom: &corev1.EnvVarSource{
-				FieldRef: &corev1.ObjectFieldSelector{
-					FieldPath: "metadata.namespace",
+	// User is responsible for knowing to restart assisted-service
+	var envFrom []corev1.EnvFromSource
+	annotations := instance.ObjectMeta.GetAnnotations()
+	configmapName, ok := annotations[configmapAnnotation]
+	if ok {
+		r.Log.Info("ConfigMap %v being used to configure assisted-service deployment", configmapName)
+		envFrom = []corev1.EnvFromSource{
```

Comment:
nit: What do you think about appending to `serviceEnv` here instead of creating another slice and calling append in L632? Maybe just rename `serviceEnv` to `envFrom` to align it with k8s terms.